pages/docs/reference/compiler-reference.md

```diff
@@ -0,0 +1,354 @@
+You can also use them for executing Kotlin script files.
+
+In addition to [common options](#common-options), Kotlin/JVM compiler has the options listed below.
+
+### `-classpath <path>` (`-cp <path>`)
+* Search for class files in the specified paths. Separate elements of the classpath with semicolons (**;**).
+The classpath can contain file and directory paths, ZIP, or JAR files.
+
+### `-d <path>`
+* Place the generated class files into the specified location. The location can be a directory, a ZIP, or a JAR file. 
+
+### `-include-runtime`
+* Include the Kotlin runtime into the resulting JAR file. Makes the resulting archive runnable on any Java-enabled 
+environment.
+
+### `-jdk-home <path>`
+* Use a custom JDK home directory to include into classpath if it differs from the default `JAVA_HOME`.
+
+### `-jvm-target <version>`
+* Specify the target version of the generated JVM bytecode. Possible values are `1.6`, `1.8`, `9`, `10`, `11`, and `12`.
+ The default value is `1.6`.
+
+### `-java-parameters`
+* Generate metadata for Java 1.8 reflection on method parameters.
+
+### `-module-name <name>`
+* Set the custom name for the generated `.kotlin_module` file.
+  
+### `-no-jdk`
+* Exclude the Java runtime from the classpath.
+
+### `-no-reflect`
+* Exclude `kotlin-reflect.jar` from the classpath.
+  
+### `-no-stdlib`
+* Exclude both `kotlin-stdlib.jar` and `kotlin-reflect.jar` from the classpath. 
+  
+### `-script <file>`
+* Evaluate a Kotlin script (`*.kts`) file.
+  
+### `-script-templates <classnames[,]>`
+* Script definition template classes. Use fully qualified class names and separate them with commas (**,**).
+
+
+## Kotlin/JS compiler options
+
+Kotlin compiler for JS compiles Kotlin source files into JavaScript code. 
+The command-line tool for Kotlin to JS compilation is `kotlinc-js`.
+
+In addition to [common options](#common-options), Kotlin/JS compiler has the options listed below.
+
+### `-libraries <path>`
+
+- Paths to Kotlin libraries with `.meta.js` and `.kjsm` files, separated by the system path separator.
+
+### `-main {call|noCall}`
+
+- Define whether the `main` function should be called upon execution.
+
+### `-meta-info`
+
+- Generate `.meta.js` and `.kjsm` files with metadata. Use this option when creating a JS library.
+
+### `-module-kind {plain|amd|commonjs|umd}`
+
+- Kind of the JS module generated by the compiler:
+    - `plain` - a plain JS module;
+    - `commonjs` - a [CommonJS](http://www.commonjs.org/) module;
+    - `amd` - an [Asynchronous Module Definition](https://en.wikipedia.org/wiki/Asynchronous_module_definition) module;
+    - `umd` - a [Universal Module Definition](https://github.com/umdjs/umd) module.
+    
+    To learn more about the module kinds and dictinctions between them,
+    read [this](https://www.davidbcalhoun.com/2014/what-is-amd-commonjs-and-umd/) article.
+    
+### `-no-stdlib`
+
+- Don't use the bundled Kotlin standard library.
```

Comment:
Sounds a bit ambiguous to me and too different from JVM's `-no-stdlib` option. Maybe something like this?
```
Exclude `kotlin-stdlib-js.jar` from the compilation dependencies.
```
(Maybe we should also change the description in `kotlinc-js -help`...)

Reply:
Yes, I'll suggest some changes to option descriptions in the Kotlin repo as well.

Reply:
As discussed with Zalim, this option works quite different in JS. Since the JS compiler implicitly adds stdlib as a compilation dependency, the most accurate description for the option is "Don't add" or "Don't use". So I rephrase this closer to the original description.